### PR TITLE
Change the error message to be meaningful naming

### PR DESCRIPTION
**Pull request by dasahcc**
### Issues

- [ ] My PR addresses the following Helix issues and references them in the PR description:

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Change the error message to be meaningful result. People see "CustomPartition:partition_1" may not understand what's going on.

### Tests

- [ ] The following tests are written for this issue:

- [x] The following is the result of the "mvn test" command on the appropriate module:

[INFO] Tests run: 160, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 53.001 s - in TestSuite
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 160, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:00 min
[INFO] Finished at: 2020-05-29T16:18:04-07:00
[INFO] Final Memory: 30M/824M
[INFO] ------------------------------------------------------------------------

### Commits

- [x] My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation (Optional)

- [ ] In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Code Quality

- [x] My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/main/java/org/apache/helix/rest/server/json/instance/StoppableCheck.java [hunk 1]
```diff
@@ -30,8 +30,8 @@
   // Category to differentiate which step the check fails
   public enum Category {
     HELIX_OWN_CHECK("Helix:"),
-    CUSTOM_INSTANCE_CHECK("CustomInstance:"),
-    CUSTOM_PARTITION_CHECK("CustomPartition:");
+    CUSTOM_INSTANCE_CHECK("CustomInstanceHealthFailure:"),
+    CUSTOM_PARTITION_CHECK("CustomPartitionHealthFailure:");
```
Comment (pkuwm) [inline helix-rest/src/main/java/org/apache/helix/rest/server/json/instance/StoppableCheck.java:34]: There are other fields in the response json as below.
```
{
  "instance_stoppable_parallel" : [ ],
  "instance_not_stoppable_with_reasons" : {
    "localhost_8080": [
        "CustomPartition:partion1",
        "CustomPartition:partition2"
    ]
  }
}

{
  "instance_stoppable_parallel" : [ ],
  "instance_not_stoppable_with_reasons" : {
    "localhost_8080" : [ "CustomInstance:localhost_8080" ]
  }
}
```

 Shall we make the keys format consistent: `custom_health_failure_instance: `? (`_instance` at the end indicates the value is an instance. And add a space after colon `:`?) It looks more consistent and clearer to me.
```
{
  "instance_stoppable_parallel" : [ ],
  "instance_not_stoppable_with_reasons" : {
    "localhost_8080": [
        "custom_health_failure_partition: partion1",
        "custom_health_failure_partition: partition2"
    ]
  }
}
```

Just a thought: not sure if we have to consider backward-compatibility: if some tools are parsing the json response to determine stoppable, this change will break the tools. Internal code search doesn't show such code usage.

Comment (dasahcc) [reply helix-rest/src/main/java/org/apache/helix/rest/server/json/instance/StoppableCheck.java]: Good suggestion.

File: helix-rest/src/main/java/org/apache/helix/rest/server/json/instance/StoppableCheck.java [hunk 2]
```diff
@@ -29,9 +29,9 @@
 public class StoppableCheck {
   // Category to differentiate which step the check fails
   public enum Category {
-    HELIX_OWN_CHECK("Helix:"),
-    CUSTOM_INSTANCE_CHECK("CustomInstance:"),
-    CUSTOM_PARTITION_CHECK("CustomPartition:");
+    HELIX_OWN_CHECK("HELIX:"),
+    CUSTOM_INSTANCE_CHECK("CUSTOM_INSTANCE_HEALTH_FAILURE:"),
```
Comment (pkuwm) [inline helix-rest/src/main/java/org/apache/helix/rest/server/json/instance/StoppableCheck.java:33]: Do you think lower case would be more consistent with other fields and easier to read? Adding a space after colon is clearer?

Comment (jiajunwang) [reply helix-rest/src/main/java/org/apache/helix/rest/server/json/instance/StoppableCheck.java:33]: I would prefer to put the ":" into the string formatting part later. The enum info should be cleaner the better.

Comment (pkuwm) [reply helix-rest/src/main/java/org/apache/helix/rest/server/json/instance/StoppableCheck.java:33]: Yeah.. I think the best way is to have a json class for this.. But that would need more work. Maybe we could do it later in helix rest improvement.

Comment (dasahcc) [reply helix-rest/src/main/java/org/apache/helix/rest/server/json/instance/StoppableCheck.java:33]: This require more factor work for the TODO as we discussed before. This change only helps make clear understanding of the error. Better to address in different PR.

**Comment by dasahcc** [comment]
This PR is ready to be merged, approved by @pkuwm

**Pull request merged**